### ISSUE TITLE
object detection Labels file format 

### ISSUE THREAD

**Issue by suji104** [NVIDIA/DIGITS#1549]
Hi. 
I want to change the label file format.
The text is shown below.

139.429527 195.206547 0.020616 11 1 0 127 151 152 128 191 192 200 199
294.314011 181.230198 0.022723 9 1 0 283 305 306 284 176 177 186 186
381.729635 126.668279 -3.101075 1 1 0 373 392 391 372 123 124 131 130
452.872890 314.469042 2.971534 2 1 0 440 465 466 442 312 308 318 322
492.282767 318.257052 0.110657 2 1 0 480 505 505 481 312 315 325 322
204.052005 444.018642 0.000000 11 1 0 192 216 216 193 440 440 448 448

I want to make this file with the following structure.
#Values    Name           Description
--------------------------------------------------------------------------------------------------------------------
   1            type             Describes the type of object: 'Car', 'Van', 'Truck', 'Pedestrian', 'Person_sitting', 
                                     'Cyclist', 'Tram', 'Misc' or 'DontCare'
   1         truncated         Float from 0 (non-truncated) to 1 (truncated), where truncated refers to the 
                                      object leaving image boundaries
   1         occluded          Integer (0,1,2,3) indicating occlusion state: 0 = fully visible, 1 = partly occluded, 
                                       2 = largely occluded, 3 = unknown
   1         alpha              Observation angle of object, ranging [-pi..pi]
   4         bbox               2D bounding box of object in the image (0-based index): contains left, top, right, 
                                      bottom pixel coordinates
   3         dimensions      3D object dimensions: height, width, length (in meters)
   3         location           3D object location x,y,z in camera coordinates (in meters)
   1         rotation_y        Rotation ry around Y-axis in camera coordinates [-pi..pi]
   1         score              Only for results: Float, indicating confidence in detection, needed for p/r curves, 
                                      higher is better.
--------------------------------------------------------------------------------------------------------------------

Below is an example.
Pedestrian 0.00 1 0.30 883.68 144.15 937.35 259.01 1.90 0.42 1.04 5.06 1.43 12.42 0.68
Pedestrian 0.00 2 0.29 873.70 152.10 933.44 256.07 1.87 0.50 0.90 5.42 1.50 13.43 0.67
Car 0.00 0 1.74 444.29 171.04 504.95 225.82 1.86 1.57 3.83 -4.95 1.83 26.64 1.55
Pedestrian 0.00 0 -0.39 649.28 168.10 664.61 206.40 1.78 0.53 0.95 2.20 1.57 34.08 -0.33
Car 0.98 0 2.42 0.00 217.12 85.92 374.00 1.50 1.46 3.70 -5.12 1.85 4.13 1.56
Pedestrian 0.00 0 2.02 240.35 190.31 268.02 261.61 1.54 0.57 0.41 -7.92 1.94 15.95 1.57
DontCare -1 -1 -10 0.00 226.06 88.58 373.00 -1 -1 -1 -1000 -1000 -1000 -10
DontCare -1 -1 -10 567.39 173.95 574.86 190.60 -1 -1 -1 -1000 -1000 -1000 -10
DontCare -1 -1 -10 727.58 165.75 737.08 192.75 -1 -1 -1 -1000 -1000 -1000 -10

How do I change it?

**Comment by sani1486** [comment]
I am looking for this solution tooo... have you figured out how to change it ?

**Comment by suji104** [comment]
No... I have not found it yet.
I have found a tool called euclid, but object detection is not working well.
Please refer to it as it may help.
https://github.com/prabindh/euclid

**Comment by sani1486** [comment]
Thanks for suggesting the tool. May i know what dataset you are using ? i have a dataset it seems the labelling format is quite same.

**Comment by sani1486** [comment]
can you tell me more about your object -detection issues ?

**Comment by suji104** [comment]
Sorry for the delay. 
I would like to use the kespry Aerial image as datasets.
What datasets do you have?
I want to train a dataset and input an arbitrary aerial image to detect the object of the aerial images.

**Comment by sani1486** [comment]
@suji104.  sorry for replying late . as far as i know , kespry datasets are not available for public use . my datasets are aerial image of vehicles. and my aim is same as yours . so have you managed to train the datasets successfully ?

**Comment by suji104** [comment]
No. Unfortunately. I do not know what the cause is. I try to try again in another way. Did you convert your dataset to the KITTI data format?

**Comment by suji104** [comment]
@sani1486 I'm afraid I'm sorry but can you answer?T.T Have you managed to train the datasets successfully? I am still having difficulties.

**Comment by sani1486** [comment]
@suji104 i stopped using that dataset . but i managed to convert to kitti label format . but not succesfull with training yet . but i can suggest you how your dataset can work , i assume they are 1024*1024 resolution . and pixel size of the objects are very small , around 40~50 pixels right ?

**Comment by suji104** [comment]
@sani1486  Thanks. Yes. We reduced the size of the image to 256x256. The size of the object in the image will be about the size of the pixel you are talking about. Is it too small to display the bounding box? And how did you convert to Kitti label format?

**Comment by RomanSteinberg** [comment]
@sani1486 I 'm sorry but I don't understand the problem. Can you write python script to transform data to the apropriate format?

**Comment by sani1486** [comment]
@suji104 
[https://alpslabel.wordpress.com/](url)
i used this tool to convert them into kitti format ,, actually you have to manually draw bounding boxes in each images and these tool will label them in kitti's format... unless you can write a python script that does it all for you .

**Comment by sani1486** [comment]
@RomanSteinberg no  i can't . but it should not be too tough , i believe.

**Comment by RomanSteinberg** [comment]
@sani1486 In my opinion DIGITS 5 is not an end-product. So, I had to look throw files writen on caffe and python to understand certain things. I believe it is extremly hard to obtain good results using DIGITS without skills in Python, Caffe and Programming.

**Comment by kargarisaac** [comment]
is there any code to convert tracklet_labels.xml file into text files?
I have downloaded data from http://www.cvlibs.net/datasets/kitti/raw_data.php and want to convert labels to files same as labels from http://www.cvlibs.net/datasets/kitti/eval_object.php?obj_benchmark=3d
